Improve UnexpectedMethodCallError output

Example:
```python
class TestOs:
    def test_getcwd(self):
        with stubout(os, "getcwd") as m_getcwd:
            pass

        # will raise a UnexpectedMethodCallError
        assert os.getcwd() == "/mox/path"
        mox.verify(m_getcwd)
```

Output:
```python
INTERNALERROR>   File "/home/user/Library/Caches/pypoetry/virtualenvs/pymox-SR0g0gTH-py3.12/lib/python3.12/site-packages/_pytest/reports.py", line 362, in from_item_and_call
INTERNALERROR>     longrepr = item.repr_failure(excinfo)
INTERNALERROR>                ^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/home/use//Library/Caches/pypoetry/virtualenvs/pymox-SR0g0gTH-py3.12/lib/python3.12/site-packages/_pytest/python.py", line 1829, in repr_failure
INTERNALERROR>     return self._repr_failure_py(excinfo, style=style)
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/home/user/Library/Caches/pypoetry/virtualenvs/pymox-SR0g0gTH-py3.12/lib/python3.12/site-packages/_pytest/nodes.py", line 481, in _repr_failure_py
INTERNALERROR>     abspath = Path(os.getcwd()) != self.config.invocation_params.dir
INTERNALERROR>                    ^^^^^^^^^^^
INTERNALERROR>   File "/home/user/projects/pymox/mox/mox.py", line 851, in __call__
INTERNALERROR>     return mock_method(*params, **named_params)
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/home/user//projects/pymox/mox/mox.py", line 1159, in __call__
INTERNALERROR>     expected_method = self._verify_method_call()
INTERNALERROR>                       ^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/home/user/projects/pymox/mox/mox.py", line 1209, in _verify_method_call
INTERNALERROR>     expected = self._pop_next_method()
INTERNALERROR>                ^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/home/user/projects/pymox/mox/mox.py", line 1195, in _pop_next_method
INTERNALERROR>     raise exception
INTERNALERROR> mox.exceptions.UnexpectedMethodCallError: Unexpected method call builtin_function_or_method.__call__() -> None
```

Similar code using Python's mock library:

```python
class TestOs:
    def test_mock(self):
        with mock.patch("os.getcwd") as m_getcwd:
            m_getcwd.return_value = '/mox/path'
            os.getcwd() == "/mox/path"

        m_getcwd.assert_not_called()
```

output:

```python
=================================== FAILURES ===================================
______________________________ TestOs4.test_mock _______________________________

self = <scratch_45.TestOs4 object at 0x10a4178c0>

    def test_mock(self):
        with mock.patch("os.getcwd") as m_getcwd:
            m_getcwd.return_value = '/mox/path'
            os.getcwd() == "/mox/path"
    
>       m_getcwd.assert_not_called()

scratch_45.py:49: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <MagicMock name='getcwd' id='4460262704'>

    def assert_not_called(self):
        """assert that the mock was never called.
        """
        if self.call_count != 0:
            msg = ("Expected '%s' to not have been called. Called %s times.%s"
                   % (self._mock_name or 'mock',
                      self.call_count,
                      self._calls_repr()))
>           raise AssertionError(msg)
E           AssertionError: Expected 'getcwd' to not have been called. Called 1 times.
E           Calls: [call()].
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -40,7 +40,7 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true'
 
       - name: Run tests and collect coverage
-        run: poetry run pytest
+        run: poetry run pytest -vv --cov=mox --cov-branch --cov-report=xml test
 
       - name: Prepare coverage token
         if: success() && github.repository == 'ivancrneto/pymox'

--- a/mox/exceptions.py
+++ b/mox/exceptions.py
@@ -63,7 +63,7 @@ class UnexpectedMethodCallError(Error):
 
         Error.__init__(self)
         if expected is None:
-            self._str = "Unexpected method call %s" % (unexpected_method,)
+            self._str = 'Unexpected method call "%s"' % (unexpected_method,)
         else:
             differ = difflib.Differ()
             diff = differ.compare(str(unexpected_method).splitlines(True), str(expected).splitlines(True))

--- a/mox/mox.py
+++ b/mox/mox.py
@@ -554,6 +554,8 @@ class MockObject(MockAnything, object):
                 if class_name == "type":
                     class_name = class_to_mock.__self__.__name__
                 self._description = "{}.{}".format(class_name, method_name)
+            elif inspect.isbuiltin(self._class_to_mock):
+                self._description = class_to_mock.__name__
             else:
                 search_string = (
                     r"<(?P<extra>function|((un)?bound )?method) (?P<class>\w*)" r"(\.(?P<method>\w*))?( at \w+)?>"
@@ -810,7 +812,7 @@ class MockObject(MockAnything, object):
 
         try:
             return mock_method(*params, **named_params)
-        except AttributeError as e:
+        except (AttributeError, Error) as e:
             Mox.unset_stubs_for_id(self._mox_id)
             raise e
 
@@ -1188,6 +1190,9 @@ class MockMethod(object):
         params = ", ".join(
             [repr(p) for p in self._params or []] + ["%s=%r" % x for x in sorted((self._named_params or {}).items())]
         )
+        if self._description and self._name == "__call__":
+            return "%s(%s) -> %r" % (self._description, params, self._return_value)
+
         full_desc = "%s(%s) -> %r" % (self._name, params, self._return_value)
         if self._description:
             full_desc = "%s.%s" % (self._description, full_desc)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -54,7 +54,7 @@ build = "^1.0.3"
 
 [tool.poetry.group.dev.dependencies]
 coverage = "^7.2"
-pytest = "^7.1"
+pytest = "^7.4"
 pytest-beeprint = "^0.1.2"
 pytest-cov = "^4.1.0"
 pytest-xdist = "^3.3.1"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -53,15 +53,11 @@ python = ">=3.8, <4"
 build = "^1.0.3"
 
 [tool.poetry.group.dev.dependencies]
-pytest = [
-    {version = "^7.1", python = ">=3.8"}
-]
-pytest-cov = [
-    {version = "^4.0", python = ">=3.8"}
-]
-coverage = [
-    {version = "^7.2", python = ">=3.8"}
-]
+coverage = "^7.2"
+pytest = "^7.1"
+pytest-beeprint = "^0.1.2"
+pytest-cov = "^4.1.0"
+pytest-xdist = "^3.3.1"
 
 [tool.hatch.version]
 source = "versioningit"
@@ -91,12 +87,9 @@ packages = ["mox", "mox/testing"]
 
 [tool.pytest.ini_options]
 addopts = [
+    "-s",
     "-vv",
-    "--cov=mox",
-    "--cov-branch",
-    "--cov-report=xml",
-    "--ignore-glob=*_helper.py",
-    "test"
+    "--ignore-glob=*_helper.py"
 ]
 python_files = "*test*.py"
 

--- a/test/mox_test.py
+++ b/test/mox_test.py
@@ -3054,7 +3054,7 @@ class MoxContextManagerTest:
         with mox.stubout("os.getcwd") as mock_obj, mox.expect:
             mock_obj().returns("/")
 
-        with pytest.raises(mox.UnexpectedMethodCallError, match='Unexpected method call "getcwd\(\) -> None"'):
+        with pytest.raises(mox.UnexpectedMethodCallError, match=r'Unexpected method call "getcwd\(\) -> None"'):
             mock_obj()
             mock_obj()
 

--- a/test/mox_test.py
+++ b/test/mox_test.py
@@ -1034,6 +1034,14 @@ class MockObjectTest(unittest.TestCase):
         )
         self.mock.reset_all()
 
+    def test_description_builtin(self):
+        mock_getcwd = self.mock.stubout("os.getcwd")
+        mock_getcwd().returns("/")
+
+        self.mock.replay_all()
+        assert mock_getcwd._description == "getcwd"
+        self.mock.reset_all()
+
     def test_mox_id(self):
         mock = mox.MockObject(TestClass)
         assert mock._mox_id is None
@@ -2423,10 +2431,9 @@ class MoxTest(unittest.TestCase):
         TestClass.other_valid_call(instance).returns("foo")
         self.mox.replay_all()
 
+        assert len(self.mox.stubs.cache) == 1
         # This should fail, since the instances are different
         self.assertRaises(mox.UnexpectedMethodCallError, TestClass.other_valid_call, "wrong self")
-
-        assert len(self.mox.stubs.cache) == 1
         self.assertRaises(mox.SwallowedExceptionError, self.mox.verify_all)
         assert len(self.mox.stubs.cache) == 0
 
@@ -3042,6 +3049,15 @@ class MoxContextManagerTest:
         with pytest.raises(mox.UnexpectedMethodCallError):
             mock_obj("ZOOBAZ")
 
+    def test_builin_with_bad_call(self):
+        """Test verifying calls to a builtin works."""
+        with mox.stubout("os.getcwd") as mock_obj, mox.expect:
+            mock_obj().returns("/")
+
+        with pytest.raises(mox.UnexpectedMethodCallError, match='Unexpected method call "getcwd\(\) -> None"'):
+            mock_obj()
+            mock_obj()
+
     def test_callable_object_verifies_signature(self):
         mock_obj = mox.create(CallableClass)
 
@@ -3517,12 +3533,13 @@ class MoxContextManagerTest:
         with mox.stubout(TestClass, "other_valid_call") as stub, mox.expect:
             TestClass.other_valid_call(instance).returns("foo")
 
+        m = mox.Mox._instances[stub._mox_id]
+        assert len(m.stubs.cache) == 1
+
         # This should fail, since the instances are different
         with pytest.raises(mox.UnexpectedMethodCallError):
             TestClass.other_valid_call("wrong self")
 
-        m = mox.Mox._instances[stub._mox_id]
-        assert len(m.stubs.cache) == 1
         with pytest.raises(mox.SwallowedExceptionError):
             m.verify_all()
         assert len(m.stubs.cache) == 0
@@ -4187,6 +4204,7 @@ class MoxTestBaseTest(unittest.TestCase):
         # Ensure no calls are made to verify_all()
         self.mox.stubout(self.test_mox, "verify_all")
         self.test_mox.unset_stubs()
+        self.test_mox.unset_stubs()
         self.test_stubs.unset_all()
         self.test_stubs.smart_unset_all()
         self.mox.replay_all()
@@ -4348,6 +4366,7 @@ class MoxTestBaseContextManagerTest(unittest.TestCase):
         m.stubout(self.test_mox, "verify_all")
 
         with m.expect:
+            self.test_mox.unset_stubs()
             self.test_mox.unset_stubs()
             self.test_stubs.unset_all()
             self.test_stubs.smart_unset_all()

